cetmix_tower_server/models/cx_tower_server_template.py

```
@@ -299,76 +311,120 @@ def _prepare_server_values(self, **kwargs):
 
                     values[field] = new_records
 
-            # custom specific variable values
+            # Handle configuration variables if provided.
             configuration_variables = kwargs.pop("configuration_variables", None)
-            line_ids_variables = kwargs.pop("line_ids_variables", None)
+            configuration_variable_options = kwargs.pop(
+                "configuration_variable_options", {}
+            )
+
             if configuration_variables:
                 # Validate required variables
                 self._validate_required_variables(configuration_variables)
 
-                variable_vals_list = []
+                # Search for existing variable options.
+                option_references = list(configuration_variable_options.values())
+                existing_options = option_references and self.env[
+                    "cx.tower.variable.option"
+                ].search([("reference", "in", option_references)])
+                missing_options = list(
+                    set(option_references)
+                    - {option.reference for option in existing_options}
+                )
+
+                if missing_options:
+                    raise UserError(
```

Question: why not to raise a validation error here? Imho this is more a code validation issue rather than a user error.
Because user can simply not have access to code. Eg when this code is triggered by an automated action.

---

ok, I will fix it